lib/graphql/batch.rb

```diff
@@ -1,6 +1,48 @@
 require "graphql"
+
+module GraphQL
+  module Batch
+    module_function
+
+    def execute(container)
+      unless container.is_a?(QueryContainer)
+        return container
+      end
+      if container.completed?
+        raise container.error if container.error
+        return container.result
+      end
+
+      batched_queries = Hash.new{ |hash, key| hash[key] = [] }
+      register_queries = lambda do |query_container|
+        query_container.each_query do |query|
+          batched_queries[query.group_key] << query
+        end
+      end
+      register_queries.call(container)
+
+      until batched_queries.empty?
+        queries = batched_queries.shift.last
```

Comment:
`last` here is used to isolate the value, because `shift` on a hash gives `[key, value]` right? is there a clearer syntax for this? (`.values.shift` is clearer but presumably doesn't actually remove the value from the hash)